Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,413 @@
+    , sosAppRuleObject = af_object // App Rule Object Name.
+    , sosAppRuleObjectContent = contentObject // App Rule Object Content.
+    , sosAppRuleAction = af_action
+    , sosSourceIPv6Address = srcV6
+    , sosDestinationIPv6Address = dstV6
+    , sosAppFullString = appcat // The full "<CATEGORY> <APPNAME> -- <SIGNAME>" string.
+    , sosAppIDNumber = app // Numeric Application ID. Not the same as "ApplicationProtocol".
+    , sosAppID = appid // Application ID from App Control
+    , sosAppCategoryID = catid // Application Category ID
+    , sosAppSignatureID = sid // Application Signature ID
+    , sosIPSCategoryName = ipscat // IPS Category Name
+    , sosAntiSpywareCategory = spycat // Anti-Spyware Category
+    , sosURLPathName = arg // URL. Represents the URL path name.
+    , sosFileIdentifier = fileid // File hash or URL
+    , sosDPIInspectedFlow = dpi // Indicates a flow was inspected by DPI. Applies only to Connection Closed messages.
+    , DstNatPortNumber = dnpt
+    , SrcNatPortNumber = snpt
+    , sosBladeID = bid // Blade ID
+    , sosUUID = uuid
+    , sosHttpRequestMethod = HttpRequestMethod
+    , sosFileName = FileName
+    , DvcOriginalAction = fw_action
+  | extend
+    ThreatName = coalesce(sosASWSignatureName, sosGAVSignatureName, sosIPSSignatureName, "")
+    , ThreatId = coalesce(sosAppSignatureID, "")
+    , ThreatCategory = coalesce(sosIPSCategoryName, sosAntiSpywareCategory, "")
+    , DstNatPortNumber = toint(DstNatPortNumber)
+    , SrcNatPortNumber = toint(SrcNatPortNumber)
+  | extend AdditionalFields = bag_pack(
+    "ReceivedPackets", sosReceivedPackets
+    , "SentPackets", sosSentPackets
+    , "ConnectionDuration", sosConnectionDuration
+    , "AppRulePolicyId", sosAppRulePolicyId
+    , "AppRulePolicyName", sosAppRulePolicyName
+    , "AppRuleService", sosAppRuleService
+    , "AppRuleType", sosAppRuleType
+    , "AppRuleObject", sosAppRuleObject
+    , "AppRuleObjectContent", sosAppRuleObjectContent
+    , "AppRuleAction", sosAppRuleAction
+    , "AppIDNumber", sosAppIDNumber
+    , "AppID", sosAppID
+    , "AppCategoryID", sosAppCategoryID
+    , "AppSignatureID", sosAppSignatureID
+    , "IPSCategoryName", sosIPSCategoryName
+    , "AntiSpywareCategory", sosAntiSpywareCategory
+    , "URLPathName", sosURLPathName
+    , "FileIdentifier", sosFileIdentifier
+    , "DPIInspectedFlow", sosDPIInspectedFlow
+    , "BladeID", sosBladeID
+    , "UUID", sosUUID
+    , "HttpRequestMethod", sosHttpRequestMethod
+    , "FileName", sosFileName
+    , "FileSize", FileSize
+    , "HttpReferrer", sosHttpReferrer
+    , "HttpUserAgent", HttpUserAgent
+    , "CaptureATPVerdict", sosCaptureATPVerdict
+    , "ApplicationID", sosApplicationID
+    , "CFSCategoryID", sosCFSCategoryID
+    , "CFSCategoryName", sosCFSCategoryName
+    , "CFSPolicyName", sosCFSPolicyName
+    , "AppControlFileName", sosAppControlFileName
+    , "AppCategoryName", sosAppCategoryName
+    , "AppIDName", sosAppIDName

```

Comment:
https://learn.microsoft.com/en-us/azure/sentinel/normalization-schema-network#destination-application-fields
or
https://learn.microsoft.com/en-us/azure/sentinel/normalization-schema-network#source-application-fields

Reply:
These 'AppCategoryName', 'AppIDName', and 'AppSignatureName' fields are created by extracting sections of the 'appcat' field. The DstAppName/SrcAppName fields are used by appcat (or appName) for the full string as sent by the reporting firewall. The intention was to make these values available in separate fields for use in Workbook queries without having to rewrite the extractions in each query.

I will just remove them...